runtime/unix.c

```diff
@@ -434,3 +435,23 @@ int caml_num_rows_fd(int fd)
   return -1;
 #endif
 }
+
+int caml_thread_setname(const char* name)
+{
+#ifdef __APPLE__
+  pthread_setname_np(name);
+  return 0;
+#else
+#ifdef _GNU_SOURCE
+  int ret;
+  pthread_t self = pthread_self();
+
+  ret = pthread_setname_np(self, name);
+  if (ret == ERANGE)
+    return -1;
+  return 0;
+#else /* not glibc, not apple */
+  return 0;
```

Comment:
just a note here (not blocker for mvp) that OpenBSD/FreeBSD also have pthread_set_name_np and NetBSD has something similar as well, so these will need to go in as well at some point.

Reply:
fix proposed in https://github.com/ocaml-multicore/ocaml-multicore/pull/829